src/aiida/engine/daemon/execmanager.py

```diff
@@ -173,25 +171,25 @@ def upload_calculation(
             # Note: this will possibly overwrite files
             for root, dirnames, filenames in code.base.repository.walk():
                 # mkdir of root
-                transport.makedirs(workdir.join(root), ignore_existing=True)
+                transport.makedirs(str(workdir.joinpath(root)), ignore_existing=True)
 
                 # remotely mkdir first
                 for dirname in dirnames:
-                    transport.makedirs(workdir.join(root, dirname), ignore_existing=True)
+                    transport.makedirs(str(workdir.joinpath(root, dirname)), ignore_existing=True)
 
                 # Note, once #2579 is implemented, use the `node.open` method instead of the named temporary file in
                 # combination with the new `Transport.put_object_from_filelike`
                 # Since the content of the node could potentially be binary, we read the raw bytes and pass them on
                 for filename in filenames:
                     with NamedTemporaryFile(mode='wb+') as handle:
-                        content = code.base.repository.get_object_content((pathlib.Path(root) / filename), mode='rb')
+                        content = code.base.repository.get_object_content((Path(root) / filename), mode='rb')
```

Comment:
```suggestion
                        content = code.base.repository.get_object_content(Path(root) / filename, mode='rb')
```
Probably also works without the brackets